pennylane_lightning/src/util/SparseLinAlg.hpp

```diff
@@ -181,52 +180,27 @@ namespace Pennylane::Util {
  * @param entries_ptr   pointer to the column indices of the non-zero elements.
  * @param values_ptr    non-zero elements.
  * @param numNNZ        number of non-zero elements.
- * @param result        result of the matrix vector multiplication.
+ * @return result        result of the matrix vector multiplication
  */
 template <class fp_precision, class index_type>
-void apply_Sparse_Matrix_Kokkos(
-    [[maybe_unused]] const std::complex<fp_precision> *vector_ptr,
-    [[maybe_unused]] const index_type vector_size,
-    [[maybe_unused]] const index_type *row_map_ptr,
-    [[maybe_unused]] const index_type row_map_size,
-    [[maybe_unused]] const index_type *entries_ptr,
-    [[maybe_unused]] const std::complex<fp_precision> *values_ptr,
-    [[maybe_unused]] const index_type numNNZ,
-    [[maybe_unused]] std::vector<std::complex<fp_precision>> &result) {
-    PL_ABORT("Executing the product of a Sparse matrix and a vector needs "
-             "Kokkos and Kokkos Kernels installation.");
+std::vector<std::complex<fp_precision>>
+apply_Sparse_Matrix(const std::complex<fp_precision> *vector_ptr,
+                    const index_type vector_size, const index_type *row_map_ptr,
+                    [[maybe_unused]] const index_type row_map_size,
+                    const index_type *entries_ptr,
+                    const std::complex<fp_precision> *values_ptr,
+                    [[maybe_unused]] const index_type numNNZ) {
+    std::vector<std::complex<fp_precision>> result;
+    result.resize(vector_size);
+    size_t count = 0;
+    for (index_type i = 0; i < vector_size; i++) {
```

Comment:
Is it possible to add OpenMP support here when ENABLE_OPENMP is set as ON?

Reply:
Yes, I think the best way would be to split the `vector_size` loop into `n_threads` blocks. Since Kokkos is always built only with a SERIAL backend and I just wanted to reproduce this functionality, I didn't add OpenMP here.